src/test/java/uk/gov/ons/ctp/response/casesvc/state/StateTransitionManagerUnitTest.java

```diff
@@ -85,4 +108,34 @@ public void testCaseTransitions() {
       });
     });
   }
+
+  @Test
+  public void testCaseGroupTransitions() {
+    StateTransitionManagerFactory factory = new CaseSvcStateTransitionManagerFactory();
+    StateTransitionManager<CaseGroupStatus, CategoryDTO.CategoryName> caseGroupStageManager =
+            factory.getStateTransitionManager(CaseSvcStateTransitionManagerFactory.CASE_GROUP);
+
+    validCaseGroupTransitions.forEach((currentState, transitions) -> {
+      transitions.forEach((caseEvent, caseState) ->{
+        try {
+          CaseGroupStatus newStatus = caseGroupStageManager.transition(currentState, caseEvent);
+          assertEquals(caseState, newStatus);
+        } catch (CTPException e) {
+          fail();
+        }
+      });
+
+      Arrays.asList(CategoryDTO.CategoryName.values()).forEach(event -> {
```

Comment:
The loop for each transaction makes this hard to follow but I appreciate it's done for consistency